pkg/ipamfetcher/config.go

```diff
@@ -0,0 +1,48 @@
+package ipamfetcher
+
+import (
+	"context"
+	"net/url"
+)
+
+// Device42ClientSettings contains configuration settings for a Device42Client
+type Device42ClientSettings struct {
+	Endpoint string
+	Limit    int
+}
+
+// Name is used by the settings library to replace the default naming convention.
+func (d *Device42ClientSettings) Name() string {
+	return "Device42Client"
+}
+
+// Device42ClientComponent satisfies the settings library Component API,
+// and may be used by the settings.NewComponent function.
+type Device42ClientComponent struct{}
+
+// Settings populates a set of default valid resource types for the Device42ClientSettings
+// if none are provided via config.
+func (d *Device42ClientComponent) Settings() *Device42ClientSettings {
+	return &Device42ClientSettings{
+		Endpoint: "",
```

Comment:
Question: Any reason why you're populating these with the default zero values?

Reply:
no reason other than the `Endpoint` is re-used for multiple components as of now, so there wasn't a default that made sense to me

Reply:
Suggestion: In that case, `return &Device42ClientSettings{}` would achieve the same effect with fewer lines.